typing/types.ml

```diff
@@ -135,36 +135,70 @@ and method_privacy =
   | Mprivate of field_kind
 
 (* Variance *)
+(* Variance forms a product lattice of the following partial orders:
+     0 <= may_pos <= pos
+     0 <= may_weak <= may_neg <= neg
+     0 <= inj
+   Additionally, the following implications are valid
+     pos => inj
+     neg => inj
+   Examples:
+     type 'a t        : may_pos + may_neg + may_weak
+     type 'a t = 'a   : pos
+     type 'a t = 'a -> unit : neg
+     type 'a t = ('a -> unit) -> unit : pos + may_weak
```

Comment:
I don't remember what `may_weak` is; is there an informal explanation of the meaning of each variance somewhere?

Reply:
The explanation is in the mli. Maybe it would be better to also move it there. However, seeing the code also helps to understand.
Specifically, `may_weak` means that there may be a non-strictly-positive occurence. Negative occurences are clearly non-strictly-positive, but any occurence under a negative one is also non-strictly-positive.
`may_weak` is only used to mark non-generalizable variables in the relaxed value restriction.